src/domain.rs

```diff
@@ -519,7 +518,7 @@ impl<F> Domain<F> {
             }
 
             if !empty {
-                crate::asymmetric_heavy_barrier(crate::HeavyBarrierKind::Expedited);
+                atomic::heavy_barrier();
```

Comment:
This doesn't seem quite right. With the "expedited" flag set, it appears the [upstream implementation](https://github.com/facebook/folly/blob/bd600cd4e88f664f285489c76b6ad835d8367cd2/folly/synchronization/AsymmetricMemoryBarrier.cpp#L86-L91) will _explicitly_ use an mprotect barrier and _not_ a `sys_membarries`.

Reply:
I think this is because folly's was written before `MEMBARRIER_CMD_PRIVATE_EXPEDITED`, membarrier falls back to mprotect if that is not available.

Reply:
Ah, interesting, thanks! Might be nice to allow non-expedited heavy barriers too, but this is good enough for me :+1: